src/Orleans.Core/Cancellation/GrainCancellationToken.cs

```diff
@@ -69,43 +64,22 @@ internal GrainCancellationToken(Guid id, bool canceled) : this(id)
 
         internal Task Cancel()
         {
-            // propagate the exception from the _cancellationTokenSource.Cancel back to the caller
-            // but also cancel _targetGrainReferences. 
-            Task task = OrleansTaskExtentions.WrapInTask(_cancellationTokenSource.Cancel);
-
-            if (_targetGrainReferences.IsEmpty)
+            if (_cancellationTokenRuntime == null)
             {
-                return task;
+                _cancellationTokenSource.Cancel();
```

Comment:
this can now throw synchronously, where as before we were returning a faulted task if that was the case

Reply:
Forgot to remove this, as it is done in the runtime part